### ISSUE TITLE
if_seq_no and if_primary_term in bulk action update

### ISSUE THREAD

**Issue by fy2** [elastic/elasticsearch-perl#184]
```perl
        $bulk->add_action(
            update => {
                id  => 123,
                doc => {
                    test_field   => "test test test",
                },
                if_seq_no       => $my_seq_no,
                if_primary_term => $my_primary_term
            },
        );
```
Is this the correct way to enter a bulk action for an update using the "if_seq_no" and "if_primary_term" parameters? I ask this because when I try it, I get an error:

`Search::Elasticsearch::Error::Unknown params <if_primary_term,if_seq_no> in <update>"`

Would you please let me know what the correct usage is?
ES version: 7.6
Search::Elasticsearch version:  6.80
Thanks.

**Comment by fy2** [comment]
Hi @ezimuel , I have found that there is a bug in a library and that's causing the problem I have described above.

The fix is:
```diff
diff --git a/lib/Search/Elasticsearch/Client/6_0/Role/API.pm b/lib/Search/Elasticsearch/Client/6_0/Role/API.pm
index db54ae40..6c3ac70e 100644
--- a/lib/Search/Elasticsearch/Client/6_0/Role/API.pm
+++ b/lib/Search/Elasticsearch/Client/6_0/Role/API.pm
@@ -42,7 +42,9 @@ sub api {
             'version'       => 'version',
             '_version'      => 'version',
             'version_type'  => 'version_type',
-            '_version_type' => 'version_type'
+            '_version_type' => 'version_type',
+            'if_primary_term' => "if_primary_term",
+            'if_seq_no'       => "if_seq_no",
         }
     },
     'bulk.update' => {
```

I have not got time to create a proper PR yet (and run the unit tests), but if I find time in the coming weeks I will give it a go.

**Comment by ezimuel** [comment]
@fy2 Right now,  `elasticsearch-perl` supports only Elasticsearch 6. We need to update all the APIs for Elasticsearch 7. I'm working now it and I'll release soon.

**Comment by fy2** [comment]
@ezimuel , I think these options are also documented for ES 6 (see "if_seq_no and if_primary_term" section here: https://www.elastic.co/guide/en/elasticsearch/reference/6.8/docs-update.html and in the bulk docs).

Thank you very much for your effort to upgrade!

**Comment by ezimuel** [comment]
@fy2 you right! [This part](https://github.com/elastic/elasticsearch-perl/blob/master/lib/Search/Elasticsearch/Client/6_0/Role/API.pm#L25-L58) of `Client/6_0/Role/API.pl` has not been updated with the ES 6.80 endpoints. I need to understand why this part is not managed by the script [build/parse_spec_6_0.pl](https://github.com/elastic/elasticsearch-perl/blob/master/build/parse_spec_6_0.pl).
I'm working on it, thanks for reporting!

**Comment by ezimuel** [comment]
@fy2 I finally fixed this in #190, sorry for the delay.

**Comment by fy2** [comment]
> @fy2 I finally fixed this in #190, sorry for the delay.

Enrico, thanks a lot for the update.

Regards, Feyruz Yalcin

**Comment by ezimuel** [comment]
Just released [6.81](https://github.com/elastic/elasticsearch-perl/releases/tag/version_6.81) version that includes this fix. Updated also CPAN here: https://metacpan.org/release/EZIMUEL/Search-Elasticsearch-6.81
